source/openpower/keys.rst

```diff
@@ -0,0 +1,148 @@
+
+If you are recieving something similar to the error::
+
+  ECDSA host key for 140.211.123.456 has changed and you have requested strict checking.
+  Host key verification failed.
+
+then a simple command should fix the error::
+
+  ~$ ssh-keygen -f "<path/to>/.ssh/known_hosts" -R <instance_IP_address>
+  
+*Example:* ``~$ ssh-keygen -f "/home/scobeavs/.ssh/known_hosts" -R 140.211.123.456``
+
+This issue occurs because the client (the machine you're attempting to log on to the server through), is also attempting to authenticate the server, to keep the user (you) from connecting to a potentially dangerous imposter. This command removes the server from your 'known_hosts' file, allowing you to connect to the server as if it were brand new to you.
+
+Changing a Key From Inside an Instance
+--------------------------------------
+
+When we create our OpenPOWER OpenStack instance, we set up a key pair to create the machine with.
+
+.. note::
+
+  For the purposes of this documentation, this key pair will be known as the **Original Key Pair**.
+
+To allow for multiple SSH Keys to access this instance:
+  We need to add the contents of ``<our_key>.pub`` to ``~/.ssh/authorized_keys`` from the machine with the corresponding keyfile.
+  
+  To do so, we need to be certain we can ``ssh`` into our instance with:: 
+  
+      ~$ ssh <image_name>@<instance_IP_address>
+    
+  After we have tested that, we can ``Ctrl+D`` or ``logout`` out of the instance, and use ``ssh-copy-id`` to transfer ``<our_key>.pub`` to ``authorized_keys``, allowing multiple key pairs.
+  
+  - To **add** a key pair, retaining the original(s)::
+  
+      ~$ ssh-copy-id <path/to/our_key>.pub <image_name>@<instance_IP_address>
+      
+    *Example:* ``~$ ssh-copy-id ~/.ssh/id_rsa.pub ubuntu@140.311.133.05``
+    
+  - To **change** the key pair, removing all others::
+
+      ~$ cat <path/to/our_key>.pub | ssh <image_name>@<instance_IP_address> "mkdir -p ~/.ssh && cat > ~/.ssh/authorized_keys"
+    
+    *Example:* ``~$ cat ~/.ssh/id_rsa.pub | ssh debian@140.244.313.10 "mkdir...``    
+    
+Changing a Key From OpenStack Web GUI
+-------------------------------------
+
+As far as in our knowledge, there is currently no way to edit the **Original Key Pair** designated to the instance.
+
+Once an instance is created, the original keypair with the original name and fingerprint will always be attached to that machine.
+
+The instance will *still* not change even if an administrator deletes the **Original Key Pair** and creates a new key pair with the same name as the **Original**, and rebuilds the machine.
+
+Our best guess to the reason why is because when the instance is created, the **Original Key Pair** and its fingerprint are entered into the ``nova`` database, and there remain, uneditable, until its destruction.
```

Comment:
Just through some research on how the instances are created, what people from openstack said, a couple docs that aren't about this but talk about how different variables are stored in the nova database upon creation.